utils/Felt.go

```diff
@@ -87,3 +90,43 @@ func FeltArrToBigIntArr(f []*felt.Felt) []*big.Int {
 	}
 	return bigArr
 }
+
+// StringToByteArrFelt converts string to array of Felt objects.
+//
+// Parameters:
+// - s: string/bytearray to convert
+// Returns:
+// - []*felt.Felt: the array of felt.Felt objects
+// - error: an error, if any
+func StringToByteArrFelt(s string) ([]*felt.Felt, error) {
+	const SHORT_LENGTH = 31
+	exp := fmt.Sprintf(".{1,%d}", SHORT_LENGTH)
+	r := regexp.MustCompile(exp)
+
+	arr := r.FindAllString(s, -1)
+	if len(arr) == 0 {
+		return []*felt.Felt{}, fmt.Errorf("invalid string no matches found, s: %s", s)
+	}
+
+	hexarr := []string{}
+	var (
+		count uint64
+		size  uint64
+	)
+
+	for _, val := range arr {
+		if len(val) == SHORT_LENGTH {
+			count += 1
+		}
+		size = uint64(len(val))
+		hexarr = append(hexarr, hex.EncodeToString([]byte(val)))
+	}
+
+	harr, err := HexArrToFelt(hexarr)
+	if err != nil {
+		return nil, err
+	}
+
+	harr = append(harr, new(felt.Felt).SetUint64(size))
```

Comment:
why are we interested in the size of the last felt?

Reply:
Based on the example as mentioned in the [doc](https://docs.starknet.io/architecture-and-concepts/smart-contracts/serialization-of-cairo-types/#serialization_of_byte_arrays)

Last element in the array indicates the size of pending word

First test case `hello` does not have any subwords with size 31. Hello will be the pending pending word of size 5
```
    0, // Number of 31-byte words in the data array.
    0x68656c6c6f, // Pending word
    5 // Length of the pending word, in bytes
 ```

Reply:
Great, thanks!